cpp/modmesh/toggle/SerializableProfiler.hpp

```diff
@@ -0,0 +1,137 @@
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <modmesh/toggle/RadixTree.hpp>
+#include <modmesh/serialization/SerializableItem.hpp>
+
+namespace modmesh
+{
+
+class SerializableRadixTreeNode : SerializableItem
+{
+
+public:
+
+    using child_list_type = std::vector<SerializableRadixTreeNode>;
+    using key_type = typename RadixTree<CallerProfile>::key_type;
+
+    SerializableRadixTreeNode() = default;
+    SerializableRadixTreeNode(SerializableRadixTreeNode const &) = default;
+    SerializableRadixTreeNode(SerializableRadixTreeNode &&) = default;
+    SerializableRadixTreeNode & operator=(SerializableRadixTreeNode const &) = default;
+    SerializableRadixTreeNode & operator=(SerializableRadixTreeNode &&) = default;
+    ~SerializableRadixTreeNode() = default;
+    SerializableRadixTreeNode(const RadixTreeNode<CallerProfile> * node)
+        : m_key(node->key())
+        , m_name(node->name())
+        , m_total_time(node->data().total_time)
+        , m_call_count(node->data().call_count)
+    {
+        for (const auto & child : node->children())
+        {
+            if (child->data().call_count > 0)
+            {
+                m_children.push_back(SerializableRadixTreeNode(child.get()));
+            }
+        }
+    }
+
+    MM_DECL_SERIALIZABLE(
+        register_member("key", m_key);
+        register_member("name", m_name);
+        register_member("total_time", static_cast<double>(m_total_time.count()));
+        register_member("call_count", m_call_count);
+        register_member("children", m_children);)
+
+private:
+
+    /*
+     * In T data, 'bool is_running' and
+     * 'std::chrono::high_resolution_clock::time_point start_time' are not serialized
+     * because they are useless when deserialize the object.
+     */
+    key_type m_key;
+    std::string m_name;
+    std::chrono::nanoseconds m_total_time;
+    int m_call_count;
+    child_list_type m_children;
+}; /* end class SerializableRadixTreeNode */
+
+class SerializableRadixTree : SerializableItem
+{
+
+public:
+
+    using key_type = typename RadixTree<CallerProfile>::key_type;
+    SerializableRadixTree() = default;
+    SerializableRadixTree(SerializableRadixTree const &) = default;
+    SerializableRadixTree(SerializableRadixTree &&) = default;
+    SerializableRadixTree & operator=(SerializableRadixTree const &) = default;
+    SerializableRadixTree & operator=(SerializableRadixTree &&) = default;
+    ~SerializableRadixTree() = default;
+    SerializableRadixTree(const RadixTree<CallerProfile> & radix_tree)
+        : m_root(radix_tree.get_root())
+        , m_id_map(radix_tree.get_stable_id_map())
+        , m_unique_id(radix_tree.get_stable_unique_node())
```

Comment:
The members are re-entrant.

Reply:
Could you please elaborate more on reentrance?  What members are related to reentrance support and how they work for it?